### PR TITLE
[FIX][14.0] website_forum: fix tag click event not found records.

### PR DESCRIPTION
**Pull request by hoangtiendung070797**
Odoo design each tag to belong to a forum. But when chosing a tag for the post if the tag already exists in another forum,
Odoo will always use that tag for the current post without creating a new tag for the forum.
Should have caused the phenomenon when clicking on a tag that is reused by the article as another forum's tag,
the system will mistakenly identify the forum and give no data or give incorrect post data.

Solution: When adding tags to posts, identify whether the new tag belongs to the current forum or not. If not on the current forum, then create new. If it belongs, then update the tag's id to the post

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by binh1702** [comment]
Eevidence:

https://user-images.githubusercontent.com/56809437/173299283-b28c5117-d1cc-4d0f-b82b-748f591c1eb9.mp4

File: addons/website_forum/controllers/main.py [hunk 1]
```diff
@@ -179,9 +179,9 @@ def forum_faq_karma(self, forum, **post):
         return request.render("website_forum.faq_karma", values)
 
     @http.route('/forum/get_tags', type='http', auth="public", methods=['GET'], website=True, sitemap=False)
-    def tag_read(self, query='', limit=25, **post):
+    def tag_read(self, query='', limit=25, forum='', **post):
```
Comment (rdeodoo) [inline addons/website_forum/controllers/main.py:182]: Just read it from the `post` kwargs? Let's not change method signature in stable
(also note that it is not `forum` but `forum_id`, i'd assume `forum` to hold a recordset)

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Thanks for your contribution.
Out of curiosity, do you know if the issue also exists in master? (And 13.0?)

File: addons/website_forum/controllers/main.py (hunk 1 above)
Comment (hoangtiendung070797) [reply addons/website_forum/controllers/main.py:182]: > Thanks for your contribution. Out of curiosity, do you know if the issue also exists in master? (And 13.0?)

@rdeodoo I tested this problem again in 13.0 and discovered that it still exists; I submitted a PR https://github.com/odoo/odoo/pull/94004 to solve it in version 13.0. Please give it a rating. Thank you very much!

Comment (hoangtiendung070797) [reply addons/website_forum/controllers/main.py:182]: @rdeodoo Don't forget me. Please !

**Comment by rdeodoo** [comment]
Will be done at https://github.com/odoo/odoo/pull/94004

**Pull request closed without merging**